mythril/laser/ethereum/transaction.py

```diff
@@ -0,0 +1,59 @@
+import logging
+from mythril.laser.ethereum.state import GlobalState, Environment, CalldataType
+from mythril.laser.ethereum.cfg import Node, Edge, JumpType
+from z3 import BitVec
+
+
+class CallTransaction:
+    """ Represents a call value transaction """
+    def __init__(self, callee_address):
+        """
+        Constructor for Call transaction, sets up all symbolic parameters
+        :param callee_address: Address of the contract that will be called
+        """
+        self.callee_address = callee_address
+        self.caller = BitVec("caller", 256)
+        self.gas_price = BitVec("gasprice", 256)
+        self.call_value = BitVec("callvalue", 256)
+        self.origin = BitVec("origin", 256)
+
+        self.open_states = None
+
+    @property
+    def has_ran(self):
+        return self.open_states is None
+
+    def run(self, open_world_states: list, evm):
```

Comment:
Does "open world states" refer to the *opening* states?

Reply:
I haven't implemented the functionality behind this part yet. My idea was to store a reference to the final world states of a transaction.
So a transaction would result in say 10 terminating states, the world states for those global states would then be referenced in the open_states variable of the transaction

Reply:
So its more like, the  "valid" states we can continue execution from

Reply:
Yeah I had indeed already thought of that ^^, that should save a lot of time